rust/lancedb/src/connection.rs

```diff
@@ -764,6 +765,110 @@ pub fn connect(uri: &str) -> ConnectBuilder {
     ConnectBuilder::new(uri)
 }
 
+/// A builder for configuring a connection to a LanceDB catalog
+#[derive(Debug)]
+pub struct CatalogConnectBuilder {
+    request: ConnectRequest,
+}
+
+impl CatalogConnectBuilder {
+    /// Create a new [`CatalogConnectBuilder`] with the given catalog URI.
+    pub fn new(uri: &str) -> Self {
+        Self {
+            request: ConnectRequest {
+                uri: uri.to_string(),
+                api_key: None,
+                region: None,
+                host_override: None,
+                #[cfg(feature = "remote")]
+                client_config: Default::default(),
+                read_consistency_interval: None,
+                storage_options: HashMap::new(),
+            },
+        }
+    }
+
+    pub fn api_key(mut self, api_key: &str) -> Self {
+        self.request.api_key = Some(api_key.to_string());
+        self
+    }
+
+    pub fn region(mut self, region: &str) -> Self {
+        self.request.region = Some(region.to_string());
+        self
+    }
+
+    pub fn host_override(mut self, host_override: &str) -> Self {
+        self.request.host_override = Some(host_override.to_string());
+        self
+    }
+
+    /// Set an option for the storage layer.
+    ///
+    /// See available options at <https://lancedb.github.io/lancedb/guides/storage/>
+    pub fn storage_option(mut self, key: impl Into<String>, value: impl Into<String>) -> Self {
+        self.request
+            .storage_options
+            .insert(key.into(), value.into());
+        self
+    }
+
+    /// Set multiple options for the storage layer.
+    ///
+    /// See available options at <https://lancedb.github.io/lancedb/guides/storage/>
+    pub fn storage_options(
+        mut self,
+        pairs: impl IntoIterator<Item = (impl Into<String>, impl Into<String>)>,
+    ) -> Self {
+        for (key, value) in pairs {
+            self.request
+                .storage_options
+                .insert(key.into(), value.into());
+        }
+        self
+    }
```

Comment:
Instead of these two can we just have one method `catalog_options` that works like `ConnectBuilder::database_options`

We will need new trait `CatalogOptions` that can look just like `DatabaseOptions`.  We don't need any implementations of `CatalogOptions` today.  Don't worry about `storage_options` today.  I will add these soon (I have an idea for what it should look like).